src/Draco.Compiler/Internal/Semantics/SemanticErrors.cs

```diff
@@ -13,29 +13,33 @@ internal static class SemanticErrors
     public static readonly DiagnosticTemplate UndefinedReference = DiagnosticTemplate.Create(
         title: "undefined reference",
         severity: DiagnosticSeverity.Error,
-        format: "undefined reference to {0}");
+        format: "undefined reference to {0}",
+        errorCode: DiagnosticTemplate.SyntaxErrorCode(2, 1));
```

Comment:
Let's not have magic constants for error categories. Let's have them explicitly defined and possibly include a factory function for them